ui-v2/src/components/ui/typography.tsx

```diff
@@ -0,0 +1,76 @@
+import { cn } from "@/lib/utils";
+import { forwardRef } from "react";
+
+type Variant = "h1" | "h2" | "h3" | "h4" | "bodyLarge" | "body" | "bodySmall";
```

Comment:
It'd be cool to follow a variant structure similar to `Button` using `cva`.

Reply:
I can look into it in a followup PR